### ISSUE TITLE
no-floating-promises misses floating promise from playwright expect chain

### ISSUE THREAD

**Issue by dmattia** [oxc-project/oxc#16408]
<img width="2354" height="1846" alt="Image" src="https://github.com/user-attachments/assets/03923496-4a20-4818-bb3c-81091b14b321" />

I'm including a picture here so that you can see that:
- The error in the console shows that I am running with type-checking, as the first example does report an error for no-reported-promises
- The second `expect(...).toBeVisible()` line returns a `Promise<void>` which should be `await`ed
- There is no reported error for the `expect(...).toBeVisible()` line

This is code coming from Playwright's testing library, via: `import { expect, test } from '@playwright/test';`, if that matters.

I am using oxlint v1.25.0 and oxlint-tsgolint v0.8.3

a minimal example file would be:

```typescript
import { expect, test } from '@playwright/test';

test('testing oxlint with playwright', async ({ page }) => {
  const thisOneDoesLintAsExpected = () => new Promise((resolve, _reject) => resolve('value'));
  thisOneDoesLintAsExpected();

  expect(
    await page.locator(`input[value="foo"]`),
  ).not.toBeVisible();
});
```

**Comment by camc314** [comment]
Thanks for reporting this, I tried to reproduce this here:

https://github.com/camc314/oxlint-repros/tree/main/issue-16408

but I could not. Could you try:
1. upgrading to the latest oxlint version
2. running with `--type-aware`? it's possible tsgo is failing to resolve something so it doesn't realize that it's a promise


Thanks
